From 7346365ce6cbfd3157fc4c3c2493c18414b7412b Mon Sep 17 00:00:00 2001
From: Michael Hanke <michael.hanke@gmail.com>
Date: Wed, 15 Sep 2021 17:06:37 +0200
Subject: [PATCH] Close file descriptors after process exit

Even a `datalad --version` leave open file descriptors around. This
change closes all FDs after a process has exited and any
protocol-based post-processing has completed.

I am not at all confident that this should be done unconditionally,
though.
---
 datalad/nonasyncrunner.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/datalad/nonasyncrunner.py b/datalad/nonasyncrunner.py
index 91544e7f4..06d463d2f 100644
--- a/datalad/nonasyncrunner.py
+++ b/datalad/nonasyncrunner.py
@@ -246,5 +246,8 @@ def run_command(cmd: Union[str, List],
     result = protocol._prepare_result()
     protocol.process_exited()
     protocol.connection_lost(None)  # TODO: check exception
+    for fd in (process.stdin, process.stdout, process.stderr):
+        if fd is not None:
+            fd.close()
 
     return result
-- 
2.33.0

